subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -404,16 +405,116 @@ static void dtls_peer_address_get(struct net_context *context,
 	memcpy(peer_addr, &context->tls->dtls_peer_addr, len);
 	*addrlen = len;
 }
+
+static inline int time_left(u32_t start, u32_t timeout)
+{
+	u32_t elapsed = k_uptime_get_32() - start;
+
+	return timeout - elapsed;
+}
+
+static int dtls_tx(void *ctx, const unsigned char *buf, size_t len)
+{
+	int sock = POINTER_TO_INT(ctx);
+	struct net_context *context = ctx;
+	ssize_t sent;
+
+	sent = zsock_sendto(sock, buf, len, context->tls->flags,
+			    &context->tls->dtls_peer_addr,
+			    context->tls->dtls_peer_addrlen);
+	if (sent < 0) {
+		if (errno == EAGAIN) {
+			return MBEDTLS_ERR_SSL_WANT_WRITE;
+		}
+
+		return MBEDTLS_ERR_NET_SEND_FAILED;
+	}
+
+	return sent;
+}
+
+static int dtls_rx(void *ctx, unsigned char *buf, size_t len, uint32_t timeout)
+{
+	int sock = POINTER_TO_INT(ctx);
```

Comment:
Could you rearrange the parameters in this function as we usually have the variables that are set a value in the beginning.